src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/enums/CFTTaskStateTest.java

```diff
@@ -33,4 +37,51 @@ void update_test_whenever_additions_to_assign_enum_are_made() {
         int assigneeEnumLength = CFTTaskState.values().length;
         assertEquals(9, assigneeEnumLength);
     }
+
+    @ParameterizedTest
+    @CsvSource(
+        value = {
+            ",",       // null
+            "''",      // empty
+            "' '",     // blank
+            "123",
+            "null",
+            "some-value"
+        }
+    )
+    void should_return_optional_empty_when_input_is_invalid(String input) {
+
+        assertEquals(Optional.empty(), CFTTaskState.from(input));
+
+    }
+
+    @ParameterizedTest
+    @CsvSource(
+        value = {
+            "UNCONFIGURED, UNCONFIGURED",
+            "unconfigured, UNCONFIGURED",
+            "PENDING_AUTO_ASSIGN, PENDING_AUTO_ASSIGN",
+            "pending_auto_assign, PENDING_AUTO_ASSIGN",
+            "ASSIGNED, ASSIGNED",
+            "assigned, ASSIGNED",
+            "CONFIGURED, CONFIGURED",
+            "configured, CONFIGURED",
+            "UNASSIGNED, UNASSIGNED",
+            "unassigned, UNASSIGNED",
+            "COMPLETED, COMPLETED",
+            "completed, COMPLETED",
+            "CANCELLED, CANCELLED",
+            "cancelled, CANCELLED",
+            "TERMINATED, TERMINATED",
+            "terminated, TERMINATED",
+            "PENDING_RECONFIGURATION, PENDING_RECONFIGURATION",
+            "pending_reconfiguration, PENDING_RECONFIGURATION",
+            "pending_reCONFIGurATion, PENDING_RECONFIGURATION"
+        }
+    )
+    void should_return_optional_empty_when_input_is_valid(String input, String expected) {
+
+        assertEquals(Optional.of(CFTTaskState.valueOf(expected)), CFTTaskState.from(input));
+
+    }
```

Comment:
It's good to cover these however I think the method `CFTTaskState.from()` is not being used anywhere so we could remove it if that's the case, but since we have the tests for it we could also leave it in case we plan of using it anyways well done

Reply:
Most of test written according to PI report. If I remove these tests coverage will decrease :)